docs/app/Components/ComponentDoc/ComponentExample.js

```diff
@@ -0,0 +1,65 @@
+import React, {Component, createElement, PropTypes} from 'react';
+import {Grid, Column, Button} from 'stardust';
+import Highlight from 'react-highlight';
+
+const req = require.context(
+  'docs/app/Examples/',
+  true,
+  /(addons|elements|collections|views|modules)\/.*\.js$/
+);
+
+/**
+ * Renders a `component` and the raw `code` that produced it.
+ * Allows toggling the the raw `code` code block.
+ */
+export default class ComponentExample extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+    description: PropTypes.string,
+    examplePath: PropTypes.string,
+    title: PropTypes.string,
+  };
+
+  state = {showCode: false};
+  fileContents = require(`!raw!docs/app/Examples/${this.props.examplePath}`);
+  component = req(`./${this.props.examplePath}.js`);
```

Comment:
We read the `examplePath` as plain text using the [`raw-loader`](https://github.com/webpack/raw-loader).  This is used to display the code in the example.

Then, we `require()` the actual component at `examplePath`.  This is used to render the example.

This is how we both show the code, and rendered example using the path to the example file.